docs/HowToUseSmartFS.md

```diff
@@ -0,0 +1,187 @@
+
+On top of the SMART MTD layer is the Smart Filesystem Layer.  
+The SmartFS code uses the logical sector services of the SMART MTD layer to
+provide file and directory level management, such as creating new files,
+chaining logical sectors together to create files, creating directories and
+ file / directory search and management routines.
+
+## FLASH Driver
+Flash driver must provide low-level read/write/erase capability for the FLASH HW.  
+Example driver is at os/arch/arm/src/s5j/s5j_sflash.c
+
+During bootup flash driver must be initialized similar to below example snippet.
+```
+void arm_boot(void)
+{
+	...
+	s5j_board_initialize();
+	...
+}
+
+void s5j_board_initialize(void)
+{
+#ifdef CONFIG_S5J_SFLASH
+        s5j_sflash_init();
+#endif
+}
+```
+### MTD Driver
+MTD driver must be enabled to access the underlying FLASH driver.  
+Example driver is at os/fs/driver/mtd/mtd_progmem.c.
+
+Below configuration must enabled for using mtd driver.
+1. Enable CONFIG_MTD  
+```
+File  Systems->Memory Technology Device (MTD) Support to y
+```
+2. Enable CONFIG_MTD_PARTITION  
+```
+File  Systems->Memory Technology Device (MTD) Support->Support MTD partitions to y
+```
+3. Enable CONFIG_MTD_PARTITION_NAMES  
+```
+File  Systems->Memory Technology Device (MTD) Support->Support MTD partition  naming to y
+```
+4. Enable CONFIG_MTD_PROGMEM  
+```
+File  Systems->Memory Technology Device (MTD) Support->Enable on-chip program  FLASH MTD device to y
+```
+
+Example initialization is as shown below,refer os/arch/arm/src/sidk_s5jt200/src/s5jt200_tash.c
+```
+artik053_configure_partitions() -> progmem_initialize ();
+```
+
+#### SMART MTD Driver
+Smart MTD layer can be enabled by enabling CONFIG_MTD_SMART config
+```
+File  Systems->Memory Technology Device (MTD) Support-> Sector Mapped  Allocation for Really Tiny (SMART) Flash support) to y
+```
+
+Refer os/fs/driver/mtd/smart.c for smart mtd layer code.
+
+smart device can be initialized using as shown below,
+```
+#if defined(CONFIG_MTD_SMART) && defined(CONFIG_FS_SMARTFS)
+                if (!strncmp(types, "smartfs,", 8)) {
+                        char partref[4];
+                        snprintf(partref, sizeof(partref), "p%d", partno);
+                        smart_initialize(CONFIG_SIDK_S5JT200_FLASH_MINOR,
+                                        mtd_part, partref);
+                }
+#endif
+```
+The above snippet creates smart device node similar to /dev/smartp0x.  
+For smart device initialization, refer os/arch/arm/src/sidk_s5jt200/src/s5jt200_tash.c
+
+##### SMART FS
+Smart FS layer can be enabled by enabling below configurations.  
+1. Enable CONFIG_FS_SMARTFS
+```
+File  Systems->SMART file system
+```
+2. Set CONFIG_SMARTFS_ERASEDSTATE
+```
+File  Systems->SMART file system ->SMARTFS options->FLASH erased state  (0xff) to 0xFF
+```
+3. set CONFIG_SMARTFS_MAXNAMLEN=32
+```
+File  Systems->SMART file system ->SMARTFS options->Maximum file length  (32) to 32
```

Comment:
is it Maximum file length or Maximum filename length ?

Reply:
Filename

Reply:
In menuconfig option is:
 Maximum file name length
We can add name before: length  (32) to 32